core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -354,8 +354,8 @@ object TopicCommand extends Logging {
 
     override def describeTopic(opts: TopicCommandOptions): Unit = {
       val topics = getTopics(opts.topic, opts.excludeInternalTopics)
-      val topicOptWithExits = opts.topic.isDefined && opts.ifExists
-      ensureTopicExists(topics, topicOptWithExits)
+      val requireTopicExists = !(opts.topic.isDefined && opts.ifExists)
```

Comment:
Initially I was going to post a comment on the redundant `opts.topic.isDefined` check here. As I was looking at this, however, I noticed that the second check seemed backwards. I think this has always been broken. We have the following documentation for the --if-exists argument:

> if set when altering or deleting or describing topics, the action will only execute if the topic exists.

But we are actually raising the exception only when `--if-exists` is not defined (i.e. `!topicOptWithExists` in the original source). I think that is actually the root of the problem. So it seems we just do the following:
```
ensureTopicExists(topics, opts.topic, opts.ifExists)
```
Does that seem right?

Reply:
No, I don't think that's right. The --if-exists essentially means "check if the topic exists, and if it does go do this". Which means, we should not be throwing an exception if the topic does not exist. It is like performing a map() operation on an Optional in Java.

If this option is not passed, AND the topic does not exist, we should be throwing an exception.

We can actually simplify line 358 above to

```
ensureTopicExists(topics, opts.topic, !opts.ifExists)
```

Let me know your thoughts.

Reply:
Ok, that makes more sense. So the expected semantic is "do nothing if the topic doesn't exist." Probably the description could be improved. Anyway, the simplification sounds good.

Reply:
@hachikuji Thanks. Updated with the simplification. The test servers seem to be having some issue connecting to github.com.

Reply:
Yeah, been seeing that lately. Not sure why.

By the way, I think I get why this logic seemed weird to me. If the topic doesn't exist, we still continue with the operation. But that doesn't make sense, right? Why go on deleting or altering the topic if it doesn't exist? Anyway, this can be improved separately.